src/darwin/Framework/CHIP/MTRDeviceControllerDelegate.h

```diff
@@ -52,8 +55,22 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
 /**
  * Notify the delegate when commissioning is completed.
  */
-- (void)controller:(MTRDeviceController *)controller commissioningComplete:(NSError * _Nullable)error;
-
+- (void)controller:(MTRDeviceController *)controller
+    commissioningComplete:(NSError * _Nullable)error MTR_NEWLY_DEPRECATED("Please use commissioningComplete:deviceId");
+/**
+ * Notify the delegate when commissioning is completed.
+ * For getting commissioning controllee's deviceId
+ */
+- (void)controller:(MTRDeviceController *)controller
+    commissioningComplete:(NSError * _Nullable)error
+                 deviceId:(NSNumber * _Nullable)deviceId MTR_NEWLY_AVAILABLE;
```

Comment:
This part is what I was thinking too.
The part that was done in the dictionary method was implemented using a dictionary because there is a possibility that it will change as the code below changes.

https://github.com/project-chip/connectedhomeip/blob/bdfe291e358161ec60900cc5e70e9f385ed4c967/src/controller/CHIPDeviceController.cpp#L2222

Would using an interface type be a better way to do it?
If implemented like that, it would be implemented like this:
`
- (void)controller:(MTRDeviceController *)controller
     readCommissioningInfo:(NSNumber *)vendorID
                 productID:(NSNumber *)productID MTR_NEWLY_AVAILABLE;
`

Reply:
It can't change by _removing_ things, right?

An interface type would look something like this:

```
@interface MTRCommissioningInfo
@property (nonatomic, copy) NSNumber * vendorID;
@property (nonatomic, copy) NSNumber * productID;
@end
```

and then you pass a `MTRCommissioningInfo *` to the delegate.